optimize/backend/src/test/java/io/camunda/optimize/service/ProcessOverviewServiceTest.java

```diff
@@ -0,0 +1,194 @@
+    tenants.add("1");
+
+    when(definitionService.getProcessDefinitionWithTenants(processKey))
+        .thenReturn(
+            Optional.of(
+                new DefinitionWithTenantIdsDto(
+                    processKey, "Process Name", PROCESS, tenants, Set.of())));
+    when(definitionAuthorizationService.isAuthorizedToAccessDefinition(any(), any(), any(), any()))
+        .thenReturn(true);
+
+    when(identityService.getUserById(newOwnerId)).thenReturn(Optional.of(new UserDto(newOwnerId)));
+    when(identityService.isUserAuthorizedToAccessIdentity(
+            userId, new IdentityDto(newOwnerId, IdentityType.USER)))
+        .thenReturn(true);
+
+    // when
+    processOverviewService.updateProcess(userId, processKey, updateDto);
+
+    // then
+    verify(processOverviewWriter).updateProcessConfiguration(processKey, updateDto);
+    verify(digestService).handleProcessUpdate(processKey, updateDto);
+  }
+
+  @Test
+  void shouldThrowExceptionWhenUnauthorizedForProcessUpdate() {
+    // given
+    final String userId = "user123";
+    final String processKey = "processKey1";
+
+    final ProcessUpdateDto updateDto =
+        new ProcessUpdateDto("owner456", new ProcessDigestRequestDto(true));
+
+    final List<String> tenants = new ArrayList<>();
+    tenants.add("1");
+
+    when(definitionService.getProcessDefinitionWithTenants(processKey))
+        .thenReturn(
+            Optional.of(
+                new DefinitionWithTenantIdsDto(
+                    processKey, "Process Name", PROCESS, tenants, Set.of())));
+    when(definitionAuthorizationService.isAuthorizedToAccessDefinition(
+            userId, PROCESS, processKey, List.of("tenant1")))
+        .thenReturn(false);
+
+    // when
+    final Throwable thrown =
+        catchThrowable(() -> processOverviewService.updateProcess(userId, processKey, updateDto));
+
+    // then
+    assertThat(thrown).isInstanceOf(ForbiddenException.class);
+  }
+
+  @Test
+  void shouldUpdateOwnerIfDefinitionIsImported() {
+    // given
+    final String userId = "user123";
+    final String processKey = "processKey1";
+    final String ownerId = "owner456";
+
+    when(definitionService.getLatestVersionToKey(PROCESS, processKey)).thenReturn("1");
+
+    when(identityService.getUserById(ownerId)).thenReturn(Optional.of(new UserDto(ownerId)));
+    when(identityService.isUserAuthorizedToAccessIdentity(
+            userId, new IdentityDto(ownerId, IdentityType.USER)))
+        .thenReturn(true);
+
+    final List<String> tenants = new ArrayList<>();
+    tenants.add("1");
+    when(definitionService.getProcessDefinitionWithTenants(processKey))
+        .thenReturn(Optional.of(new DefinitionWithTenantIdsDto(tenants)));
+
+    when(definitionAuthorizationService.isAuthorizedToAccessDefinition(any(), any(), any(), any()))
+        .thenReturn(true);
+
+    // when
+    processOverviewService.updateProcessOwnerIfNotSet(userId, processKey, ownerId);
```

Comment:
missing tests:

- Owner ID is empty or null
- Happy path if definition has been imported. We can assert on the logs here too, and the method calls
- Saving the pending update to the writer. Again, we can assert on the logs and method calls

Reply:
We can assert on the logs here too - for asserting logs we should add additional dependency I propose check only methods

Reply:
fixed